SpatialGDK/Source/SpatialGDK/Private/EngineClasses/SpatialMultiWorkerSettings.cpp

```diff
@@ -0,0 +1,178 @@
+	{
+		ValidateLockingPolicyIsSet();
+	}
+};
+#endif
+
+uint32 UAbstractSpatialMultiWorkerSettings::GetMinimumRequiredWorkerCount() const
+{
+	uint32 WorkerCount = 0;
+
+	for (const FLayerInfo& LayerInfo : WorkerLayers)
+	{
+		check(*LayerInfo.LoadBalanceStrategy != nullptr);
+		WorkerCount += GetDefault<UAbstractLBStrategy>(*LayerInfo.LoadBalanceStrategy)->GetMinimumRequiredWorkers();
+	}
+
+	return WorkerCount;
+}
+
+void UAbstractSpatialMultiWorkerSettings::ValidateNonEmptyWorkerLayers()
+{
+	if (WorkerLayers.Num() == 0 )
+	{
+		WorkerLayers.Emplace(SpatialGDK::DefaultLayerInfo);
+		FMessageDialog::Open(EAppMsgType::Ok,
+			FText::Format(LOCTEXT("EmptyWorkerLayer_Prompt", "You need at least one layer in your settings. "
+			"Adding back the default layer. File: {0}"), FText::FromString(GetNameSafe(this))));
+	}
+}
+
+void UAbstractSpatialMultiWorkerSettings::ValidateSomeLayerHasActorClass()
+{
+	bool bHasTopLevelActorClass = false;
+	for (const FLayerInfo& Layer : WorkerLayers)
+	{
+		bHasTopLevelActorClass |= Layer.ActorClasses.Contains(AActor::StaticClass());
+	}
+
+	if (!bHasTopLevelActorClass)
+	{
+		WorkerLayers[0].ActorClasses.Add(AActor::StaticClass());
+		FMessageDialog::Open(EAppMsgType::Ok,FText::Format(LOCTEXT("MissingActorLayer_Prompt",
+			"Some worker layer must contain the root Actor class. Adding AActor to the first worker layer entry. "
+			"File: {0}"), FText::FromString(GetNameSafe(this))));
+	}
+}
+
+void UAbstractSpatialMultiWorkerSettings::ValidateNoActorClassesDuplicatedAmongLayers()
+{
+	TSet<TSoftClassPtr<AActor>> FoundActorClasses{};
+	TSet<TSoftClassPtr<AActor>> DuplicatedActorClasses{};
+
+	for (FLayerInfo& Layer : WorkerLayers)
+	{
+		for (const TSoftClassPtr<AActor> LayerClass : Layer.ActorClasses)
+		{
+			if (FoundActorClasses.Contains(LayerClass))
+			{
+				DuplicatedActorClasses.Add(LayerClass);
+			}
+			FoundActorClasses.Add(LayerClass);
+		}
+
+		for (const TSoftClassPtr<AActor> DuplicatedClass : DuplicatedActorClasses)
+		{
+			Layer.ActorClasses.Remove(DuplicatedClass);
+		}
+	}
+
+	if (DuplicatedActorClasses.Num() > 0)
+	{
+		FString DuplicatedActorsList = "";
+		for (const TSoftClassPtr<AActor> DuplicatedClass : DuplicatedActorClasses)
+		{
+			DuplicatedActorsList.Append(FString::Printf(TEXT("%s, "), *DuplicatedClass.GetAssetName()));
+		}
+		FMessageDialog::Open(EAppMsgType::Ok,FText::Format(LOCTEXT("MultipleActorLayers_Prompt",
+			"Defining the same Actor type across multiple layers is invalid. Removed all occurences after the first. "
+            "File: {0}. Duplicate Actor types: {1}"), FText::FromString(GetNameSafe(this)), FText::FromString(DuplicatedActorsList)));
+	}
+}
+
+void UAbstractSpatialMultiWorkerSettings::ValidateAllLayersHaveUniqueNonemptyNames()
+{
+	TSet<FName> FoundLayerNames{};
+	bool bSomeLayerNameWasChanged = false;
+
+	uint32 LayerCount = 1;
+
+	// We currently rely on this for rendering debug information.
+	WorkerLayers[0].Name = SpatialConstants::DefaultLayer;
```

Comment:
This seems a strange spot to be setting this

Reply:
Separated out to its own validation function